### PR TITLE
Add missing arguments to construct s3 client

### PR DESCRIPTION
**Pull request by nhuray**
## Summary & Motivation

Hi community,

This is a very small PR related to `dagster-aws` library and specifically to the `s3` resource.

I wanted for local development to use the `s3_resource` with [minio](https://min.io/). But because on local I didn't configured Minio with SSL I couldn't use the `s3_resource` because some arguments passed to `boto` are not exposed.

So in that PR I just exposed `use_ssl` and `verify` args in the `construct_s3_client` function  and set default value to `use_ssl=True` and `verify=None`.

## How I Tested These Changes

I did not but it's a pretty straightforward change cause I'm just passing arguments to boto.

### CODE REVIEW

File: python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py
Comment (dpeng817) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py:33]: Probably going to need to run `make black` for this line to format properly

**Review by dpeng817 — DISMISSED** [review]
Probably fine given that it's not changing any defaults, just a few things to resolve / clean up

File: python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py
Comment (nhuray) [reply python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py:33]: Absolutely ! Good catch !

**Comment by nhuray** [comment]
@dpeng817 I changed `verify` to a string and set the default to `None`.

But I have a question related to the arguments we should expose because I notice the boto `resource` has arguments not exposed in the `construct_s3_client`:
        - aws_access_key_id
        - aws_secret_access_key
        - aws_session_token
        - config

Should we expose them considering they can also be set as environment variable (e.g  `AWS_ACCESS_KEY_ID` ) ?

**Comment by nhuray** [comment]
@dpeng817 Let me know if there's something missing to merge that PR !

Thanks

**Comment by nhuray** [comment]
> @dpeng817 I changed `verify` to a string and set the default to `None`.
> 
> But I have a question related to the arguments we should expose because I notice the boto `resource` has arguments not exposed in the `construct_s3_client`:
> 
> * aws_access_key_id
> * aws_secret_access_key
> * aws_session_token
> * config
> 
> Should we expose them considering they can also be set as environment variable (e.g `AWS_ACCESS_KEY_ID` ) ?

Answering my own question here ! I think we should expose those params in the `construct_s3_client` because we can have use cases (and it's mine) to have a code location who requires to access multiple S3 buckets but using different IAM credentials.

For instance:

```
# S3 resource for Bucket 1 
s3_bucket_1 = s3_resource.configured(
            {
                "aws_access_key_id": {"env": "BUCKET_1_AWS_ACCESS_KEY_ID"},
                "aws_secret_access_key": {"env": "BUCKET_1_AWS_SECRET_ACCESS_KEY"}
            }
        )

# S3 resource for Bucket 2 
s3_bucket_2 = s3_resource.configured(
            {
                "aws_access_key_id": {"env": "BUCKET_2_AWS_ACCESS_KEY_ID"},
                "aws_secret_access_key": {"env": "BUCKET_2_AWS_SECRET_ACCESS_KEY"}
            }
        )
```

**Comment by dpeng817** [comment]
https://buildkite.com/dagster/dagster/builds/52602 (build for maintainers to check out - unfortunately we don't have an easy way of external contributors checking this out quite yet)

File: python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py
Comment (dpeng817) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py:48]: probably want to check parameterization on last three args here as well

**Review by dpeng817 — COMMENTED** [review]
Given that the scope has increased want to loop in @sryza - any thoughts on adding these args?

File: python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py
Comment (nhuray) [reply python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py:48]: good catch. Sorry I forgot that

**Comment by nhuray** [comment]
Hi @dpeng817 @yuhan,

Is there something I can do to make that PR approved ?

**Review by yuhan — APPROVED** [review]

**Comment by yuhan** [comment]
Hey @nhuray apologies for the delay. The additional args look reasonable. I'm kicking off the CI now. Will merge when it's green :)

File: python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py
Comment (nhuray) [reply python_modules/libraries/dagster-aws/dagster_aws/s3/utils.py]: Done !

**Comment by nhuray** [comment]
@dpeng817 Let me know if some stuff are missing now to merge that PR !

Thanks

**Comment by nhuray** [comment]
@dpeng817 Let me know if some stuff are missing now to merge that PR !

Thanks

**Comment by yuhan** [comment]
@nhuray im kicking off another CI build. will make the test green and merge right after that.

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:53]:
```suggestion
       StringSource,
```
we probably want it to be `StringSource` ([docs](https://docs.dagster.io/_apidocs/config#dagster.StringSource)) so users can pass in either the string value or the env var key.

see best practices of handling secrets: https://docs.dagster.io/guides/dagster/using-environment-variables-and-secrets#handling-secrets

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:67]: same - `StringSource`

**Review by yuhan — COMMENTED** [review]
one last suggestion before we merge :)

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (nhuray) [reply python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:53]: @yuhan make sense ! Fixing it soon !

**Comment by nhuray** [comment]
@yuhan please check the last commit [here](https://github.com/dagster-io/dagster/pull/12896/commits/551b938f7484763faeb5b3f0cdf473d5d5748463) for the change I made to use `StringSource`, `BoolSource` and `IntSource` on all the config !

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:10]: i dont think you'd need to use `*Source` for it because they won't be secrets nor set in env vars

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:16]: same - `str` should be just fine

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:24]: same - `int` should be just fine

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:33]: same - `str` should be just fine

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:38]: same - `bool` should be just fine

File: python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py
Comment (yuhan) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py:44]: same - `str` should be just fine

**Review by yuhan — COMMENTED** [review]
i think we only need `aws_access_key`, `aws_secret_access_key`, and `aws_session_token` to be `StringSource` because other configs are not sensitive info which likely won't be in env vars. since other parts are looking good, im happy to make those changes to speed up this review if you're ok with it.

**Comment by nhuray** [comment]
@yuhan changes reverted in the last commit

**Comment by yuhan** [comment]
@nhuray thanks - i pushed a fix commit to fix format and fix the failing tests (i.e. make the fields `Noneable` to accommodate the `default_value=None` setting).

**Pull request merged**
